test/test_filepost.py

```diff
@@ -1,15 +1,49 @@
-import unittest
-
-from urllib3.filepost import encode_multipart_formdata, iter_fields
+import psutil
+import io
+import sys
+
+if sys.version_info >= (2, 7):
+    import unittest
+else:
+    import unittest2 as unittest
+
+from urllib3.filepost import (
+    encode_multipart_formdata,
+    iter_fields,
+    MultipartEncoderGenerator
+)
 from urllib3.fields import RequestField
 from urllib3.packages.six import b, u
 
 
 BOUNDARY = '!! test boundary !!'
 
 
-class TestIterfields(unittest.TestCase):
+class TestMultipartEncoderGenerator(unittest.TestCase):
+    def test_len(self):
+        fieldsets = [
+            [('k', 'v'), ('k2', 'v2')],
+            [('k', b'v'), (u('k2'), b'v2')],
+            [('k', b'v'), (u('k2'), 'v2')],
+            [('foo', b'a'), ('foo', b'b')],
+        ]
+
+        for fields in fieldsets:
+            encoded, _ = encode_multipart_formdata(fields, boundary=BOUNDARY)
+            encoded = b''.join(encoded)
+            self.assertEqual(len(encoded), len(MultipartEncoderGenerator(fields, boundary=BOUNDARY)))
```

Comment:
Isn't this basically testing against itself? I'd prefer to have fixed length numbers to test against.